core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-filter.model.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Model class for creating filters of frontend contributor
+ * admin Dashboard.
+ */
+export class ContributorAdminDashboardFilter {
+  languageCode?: string;
+  topicIds?: string[];
+  sort?: string;
+  lastActivity?: number;
```

Comment:
Why is this optional?

Reply:
These are supposed to be optional cause will making the request to the handler, we only have to pass filters which are set otherwise we are not supposed to pass them. This is partially implemented, it will make more sense once we implement filtering and sorting in frontend as well.